### ISSUE TITLE
recursive folder synchronization is quite slow when there are lots of files.

### ISSUE THREAD

**Issue by zspencer** [andsens/homeshick#152]
Apologies for the poorly worded title; I'm not sure exactly what I'm asking for here, so feel free to tell me to come back when I have a clearer issue.

I am using homeshick to link my .atom/packages directory between machines. It's quite nice! However... atom packages are npm packages with thousands of recursive node_modules folders!

When I have about a dozen packages installed, this results in ~4+ minute `homeshick link` calls; as there are just so many files and folders to verify are linked correctly.

I would absolutely love it if I could somehow tell homeshick to just link the entire directory instead of trying to link all the files. 

I gandered at [the source](https://github.com/andsens/homeshick/blob/master/lib/commands/link.sh#L61-L65); and assumed that this is what it's supposed to be doing? But my BASH reading comprehension is about at a 4 year old level; so anything beyond "Cow go moo/duck go quack" is a bit beyond me and I could be mistaken in how symlinking of directories is intended to work.

**Comment by nielssorensen** [comment]
I'm pretty sure you should leave your npm (your node package manager) to manage your packages and homeshick to manage your dotfiles (generally configuration files).  One tool to do one job another to do another.  There is a reason for such delineation but someone else can correct me if I'm wrong.

**Comment by andsens** [comment]
@nielssorensen 
> I'm pretty sure you should leave your npm (your node package manager) to manage your packages and homeshick to manage your dotfiles

There's some truth to that. Tracking files which can be downloaded from somewhere else fully automatically might not make sense.

> I would absolutely love it if I could somehow tell homeshick to just link the entire directory instead of trying to link all the files.

[You can totally do that](https://github.com/andsens/homeshick/wiki/Symlinking#shallow-symlinking) :-)

> But my BASH reading comprehension is about at a 4 year old level; so anything beyond "Cow go moo/duck go quack" is a bit beyond me

Haha, man I got to remember this analogy!

Anywho, I think the question is answered. Feel free to reopen if you disagree.

**Comment by zspencer** [comment]
Re: Shallow Linking - This is exactly what I want! Thanks a ton!


Re: Using a package manager - I totally agree! And for system-level dependencies I use a script `ensure-consistent-packages` that installs stuff with brew/npm/gem/etc. 

For editors like sublime/vim/emacs; the "package manager" is to clone X repo and put it in Y directory; which is perfect for homeshick. The issue only really arises with atom because atom packages are npm packages; but installed in a particular location.  Turns out there is a command line tool `apm` which I could call from my `ensure-consistent-packages` script; but that would require me to manually update the script when I add a package via Atom's built in package manager.

All told, shallow linking does *exactly* what I want, and cuts run time of `homeshick link` down to 10~20s again from the ~4m it was taking when I linked each file in `.atom/packages` directly.
